be/src/fs/fs_starlet.cpp

```diff
@@ -119,40 +154,53 @@ class StarletFileSystem : public FileSystem {
 
     Type type() const override { return STARLET; }
 
+    StatusOr<Configuration> get_shard_config(int64_t tablet_id) {
+        Configuration conf;
+        ASSIGN_OR_RETURN(auto shardinfo, g_worker->get_shard_info(tablet_id));
+        conf[kS3AccessKeyId] = shardinfo.properties[kS3AccessKeyId];
+        conf[kS3AccessKeySecret] = shardinfo.properties[kS3AccessKeySecret];
+        conf[kS3OverrideEndpoint] = shardinfo.properties[kS3OverrideEndpoint];
+        conf[kS3Bucket] = shardinfo.properties[kS3Bucket];
+        return conf;
```

Comment:
shard info should contains storage_path as well which will be `s3://bucket/path/to/tablet` this is the root path for the tablet, or whatever URL, `hdfs://nn:port/path/to/tablet`, conf[kSysRoot] will be set to this storage path, all the relative path operation to the filesystem will be related to this storage path.

Reply:
we will implement  a fs cache in next pr.